dialogue-core/src/main/java/com/palantir/dialogue/core/DialogueChannel.java

```diff
@@ -27,70 +27,40 @@
 import com.palantir.dialogue.Endpoint;
 import com.palantir.dialogue.Request;
 import com.palantir.dialogue.Response;
-import com.palantir.logsafe.Preconditions;
 import com.palantir.logsafe.Safe;
 import com.palantir.logsafe.SafeArg;
-import com.palantir.logsafe.exceptions.SafeIllegalStateException;
-import com.palantir.logsafe.exceptions.SafeRuntimeException;
-import com.palantir.random.SafeThreadLocalRandom;
-import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
 import java.util.Collection;
 import java.util.HashSet;
 import java.util.Map;
-import java.util.Optional;
 import java.util.Random;
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.atomic.AtomicReference;
-import java.util.function.Supplier;
-import javax.annotation.Nullable;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public final class DialogueChannel implements Channel {
     private static final Logger log = LoggerFactory.getLogger(DialogueChannel.class);
 
-    private final Map<String, LimitedChannel> limitedChannelByUri = new ConcurrentHashMap<>();
-    private final AtomicReference<LimitedChannel> nodeSelectionStrategy = new AtomicReference<>();
-    private final QueuedChannel queuedChannel; // just so we can process the queue when uris reload
-
-    private final String channelName;
-    private final ClientConfiguration clientConfiguration;
-    private final ChannelFactory channelFactory;
     private final Channel delegate;
-    private final ClientMetrics clientMetrics;
-    private final DialogueClientMetrics dialogueClientMetrics;
-    private final Random random;
-    private Ticker ticker;
 
-    // TODO(forozco): you really want a refreshable of uri separate from the client config
-    private DialogueChannel(
-            String channelName,
-            ClientConfiguration clientConfiguration,
-            ChannelFactory channelFactory,
-            Random random,
-            Supplier<ScheduledExecutorService> scheduler,
-            int maxQueueSize,
-            Ticker ticker) {
-        this.channelName = channelName;
-        this.clientConfiguration = clientConfiguration;
-        this.channelFactory = channelFactory;
-        clientMetrics = ClientMetrics.of(clientConfiguration.taggedMetricRegistry());
-        dialogueClientMetrics = DialogueClientMetrics.of(clientConfiguration.taggedMetricRegistry());
-        this.random = random;
-        this.ticker = ticker;
+    // we keep around internals purely for live-reloading
+    private final Config cf;
+
+    private final QueuedChannel queuedChannel; // just so we can process the queue when uris reload
+    private final Map<String, LimitedChannel> limitedChannelByUri = new ConcurrentHashMap<>();
+    private final AtomicReference<LimitedChannel> nodeSelectionChannel = new AtomicReference<>();
```

Comment:
I think having fewer fields helps emphasise what's immutable and what we end up mutating.